spec/consensus/gossip/crdt.md

```diff
@@ -130,48 +129,178 @@ Consider the following definition for **Eventual Convergence**.
 
 |Eventual Convergence|
 |-----|
-| If there exists a correct process $p$ such that $e \in t_p$, then, eventually, for every correct process $q$, $e \in t_q$ or there exists a correct process $r$ such that $\bar{e} \in t_r$.|
-
-> **Note**
-> Nodes may learn of an entry deletion before learning of its addition.
+| If there exists a correct process $p \in P$ such that $e \in t_p$, then, eventually, for every correct process $q \in P$, either $e \in t_q$ or $e$ is stale in $t_q$.
 
-In order to ensure convergence even in the presence of failures, the network must be connected in such a way to allow communication around any malicious nodes and to provide paths between correct ones.
-This can be achieved if there is a GST, after which timeouts eventually do not expire precociously, given that they all can be adjusted to reasonable values, which implies that all communication will eventually happen timely, which implies that the tuple space will converge and keep converging.
+In order to ensure convergence even in the presence of failures, the network must be connected in such a way to allow communication around any malicious nodes, that is, to provide paths connecting correct nodes.
+Even if paths connecting correct nodes exist, effectively using them requires timeouts to not expire precociously and abort communication attempts.
+Timeout values can be guaranteed to eventually be enough for communication after a GST is reached, which implies that all communication between correct processes will eventually happen timely, which implies that the tuple space will converge and keep converging.
 Formally, if there is a GST then following holds true:
 
 | Eventual $\Delta$-Timely Convergence |
 |---|
-| If $e\in t_p$, for some correct process $p$, at instant $t$, then by $\text{max}(t,\text{GST}) + \Delta$, either $e \in t_q$, for every correct process $q$ or $e$ is stale in \bar{e} \in t_p$.
+| If $e\in t_p$, for some correct process $p \in P$, at instant $t$, then by $\text{max}(t,\text{GST}) + \Delta$, for every correct process $q \in P$, either $e \in t_q$ or $e$ is stale in $t_p$.
```

Comment:
then _the_ following holds true

I am not sure to understand the benefit of $\Delta$-timely convergence over eventual converge, as the later suffices.

Reply:
hmmm... the important difference at the time of the writing was that e may get stale on q, in eventual convergence, but in eventual delta timely convergence it can get stale in $p$, and never forwarded to $q$, but I will have to think more about it being actually required or not.